deeppavlov/core/trainers/nn_trainer.py

```diff
@@ -297,5 +351,14 @@ def train(self, iterator: DataLearningIterator) -> None:
         else:
             log.warn(f'Using {self.__class__.__name__} for a pipeline without batched training')
 
-        if not self._saved:
+        # Run the at-train-exit model-saving logic
+        if self.validation_number < 1:
+            log.info('Save model to capture early training results')
             self.save()
+        else:
+            if self.improved(self.score_best, self.score_baseline):
+                log.info('Save improved model')
+                self.save()
```

Comment:
So if we had multiple validations, best score is better than a baseline, but the latest score is not the best, the latest model will be saved?

Reply:
OK, probably it is incorrect but it looks that I don't get right your last requirement

> But we need to remember to not save at the end if all validations after 0-th did not improve.

Are there any conditions when we want to save models at-exit after 1st validation?